__tests__/html/renderActivity.performance.html

```diff
@@ -0,0 +1,107 @@
+              <small> Rendered {timesActivityRendered.get(activity.id)} times</small>
+            </>
+          )
+        }
+      }
+
+      let shownCount = 0;
+      async function postMessagesBatch(directLine) {
+        const promises = [];
+        const timestamp = new Date().toISOString();
+        for (let index = 0; index < BATCH_SIZE; index++) {
+          promises.push(
+            // Plain text message isolate dependencies on Markdown.
+            directLine.emulateIncomingActivity(
+              { id: `activity-${shownCount + index}`, text: `Message ${shownCount + index}.`, textFormat: 'plain', type: 'message', timestamp },
+              { skipWait: true }
+            )
+          );
+        }
+        shownCount += BATCH_SIZE;
+
+        await Promise.all(promises);
+        await pageConditions.numActivitiesShown(shownCount);
+      }
+
+      run(
+        async function () {
+          const {
+            WebChat: { ReactWebChat }
+          } = window; // Imports in UMD fashion.
+
+          const { directLine, store } = testHelpers.createDirectLineEmulator();
+
+          WebChat.renderWebChat({ directLine, store, activityMiddleware: [activityRendered] }, document.querySelector('main'));
+
+          await pageConditions.uiConnected();
+          pageElements.transcript().focus();
+
+          // WHEN: adding 10 activities
+          await postMessagesBatch(directLine);
+          await postMessagesBatch(directLine);
+
+          // THEN: do not re-render each activity each time
+          expect(Math.max(...timesActivityRendered.values())).toEqual(2);
+          expect(Math.min(...timesActivityRendered.values())).toEqual(1);
+
+          // WHEN: clicking on an activity
+          pageElements.activities()[4].scrollIntoView();
+          await host.clickAt(10, 10, pageElements.activities()[4]);
+
+          // THEN: set focus to the activity
+          expect(pageElements.focusedActivity()).toEqual(pageElements.activities()[4]);
+          
+          // THEN: no re-renders happen
+          expect(Math.max(...timesActivityRendered.values())).toEqual(2);
+          expect(Math.min(...timesActivityRendered.values())).toEqual(1);
+
+          // WHEN: updating an activity
+          const timestamp = new Date().toISOString();
+          const activity9Renders = timesActivityRendered.get('activity-8');
+          await directLine.emulateIncomingActivity(
+              { id: `activity-8`, text: `Activity 8 got updated`, textFormat: 'plain', type: 'message', timestamp },
+              { skipWait: true }
+            );
+
+          // THEN: activity re-renders the activity
+          expect(timesActivityRendered.get('activity-8')).toBe(activity9Renders + 1);
+          // THEN: activity gets updated on screen
+          pageElements.focusedActivity().scrollIntoView();
+          await host.snapshot();
+        },
+        { skipCheckAccessibility: true }
```

Comment:
We should keep check accessibility unless we know something would definitely break.